content/en/docs/tasks/job/indexed-parallel-processing-static.md

```diff
@@ -0,0 +1,166 @@
+
+1. **Create an image that can read the pod index**
+2. **Start an Indexed Job**. The downward API allows you to pass the annotation
+   as an environment variable or file to the container.
+
+<!-- steps -->
+
+## Create an Image
+
+To access the work item from the worker program, you have a few options:
+
+1. Read the `JOB_COMPLETION_INDEX` environment variable. The Job controller
+   automatically links this variable to the annotation containing the completion
+   index.
+1. Read a file that contains the completion index.
+1. Assuming that you can't modify the program, you can wrap it with a script
+   that reads the index using any of the methods above and converts it into
+   something that the program can use as input.
+  
+In this example, we show option 3. The following program knows how to
+accept a work item from a command line argument.
+
+{{< codenew language="python" file="application/job/worker.py" >}}
+
+Give the script execution permission:
+
+```shell
+chmod +x worker.py
+```
+
+Now, build an image. Download the [Dockerfile](/examples/application/job/Dockerfile)
+and build the image with this command:
+
+```shell
+docker build -t job-worker .
+```
+
+For the [Docker Hub](https://hub.docker.com/), tag your app image with
+your username and push to the Hub with the below commands. Replace
+`<username>` with your Hub username.
+
+```shell
+docker tag job-worker <username>/job-worker
+docker push <username>/job-worker
+```
+
+If you are using [Google Container Registry](https://cloud.google.com/tools/container-registry/),
+tag your app image with your project ID, and push to GCR. Replace `<project>`
+with your project ID.
+
+```shell
+docker tag job-wq-1 gcr.io/<project>/job-worker
+gcloud docker -- push gcr.io/<project>/job-worker
+```
+
+## Define an Indexed Job
+
+Here is a job definition. You'll need to edit the image to match name you used.
+
+{{< codenew language="yaml" file="application/job/indexed-job.yaml" >}}
+
+In this example, we used the builtin `JOB_COMPLETION_INDEX` environment variable
+set by the job controller. Optionally, you can [define your own environment
+variable via through the downward API](/docs/tasks/inject-data-application/environment-variable-expose-pod-information/):
+
+{{< codenew language="yaml" file="application/job/indexed-job-env.yaml" >}}
+
+If, instead, the worker program knows how to read a workitem from a file, you can
+similarly [use the downward API to pass the annotation value as a volume file](/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#store-pod-fields).
+
+If the worker program expects certain values (for example, filenames) instead of
+obscure completion indexes, you can do the mapping in a simple script.
+
+{{< codenew language="yaml" file="application/job/indexed-job-args.yaml" >}}
+
+You can also load the list of values from a [ConfigMap as an environment variable
+or file](/docs/tasks/configure-pod-container/configure-pod-configmap/).
```

Comment:
Maybe link to the init container concept here?

Reply:
You can do it without an init container, so it seems tangential to me.